sklearn/manifold/t_sne.py

```diff
@@ -706,22 +717,41 @@ def _fit(self, X, skip_num_points=0):
         if self.n_iter < 250:
             raise ValueError("n_iter should be at least 250")
 
+        if self.square_distance not in [True, False, 'legacy', 'warn']:
+            raise ValueError("square_distance should be True, False, 'legacy'"
+                             "or warn")
+
+        if self.square_distance == 'warn':
+            if self.metric != "euclidean":
+                warnings.warn("The default value of square_distance will be"
+                              "set to True in 0.23.", FutureWarning)
+            self.square_distance = True
```

Comment:
We avoid changing the user's setting. Instead you can use a local variable or just always check for 'warn'
Besides, this is incorrect behaviour: for backwards compatibility, the default behaviour should be equivalent to 'legacy'